docs/examples/queries/entrants-by-tournament.md

```diff
@@ -0,0 +1,499 @@
+---
+id: entrants-by-tournament
+title: Entrants within a Tournament
+---
+
+## Disimbiguation
+
+It can be very easy to conflate the entities of User, Player, Participant, Entrant, and Team when querying data. A useful clarification is that a User, Player, and GlobalTeam exist in the **global context** of start.gg -- these are entities that are up-to-date on what settings a user currently has applied to their profile.
+
+A Participant, Entrant, or Team are **point-in-time** references to a User/Player/GlobalTeam. These entities are created when a User registers for a Tournament or for an Event. This means that if a player updates their `gamerTag` after a tournament has completed, the tournament's `Entrant` and `Participant` will still show the prior gamerTag at the **time of the event (or the time of event registration)**. 
+
+However, if you look at the `Player` entity, it will show the **currently set** gamerTag of a User. `User`, you may note, does not have a field for `gamerTag` or for `prefix`, as those only exist in the context of a *Player*, not a generic user of Start.GG. 
+
+When querying data about a person, such as gender pronouns, it is best to dig from the Tournament's Participants to each User entity at the time of the tournament to ensure that the data is as up-to-date as possible.
+
+### Prefix 
+
+Participants in a Tournament can edit their Prefix for a given Tournament without necessarily updating their global prefix. In this case, since the user explicitly has set what prefix they want to go with for that tournament, it may be preferable to look at the Participant entity rather than the Player entity. 
```

Comment:
This is generally another "point-in-time" consideration. The player's prefix and gamertag are what they are for the tournament in question. As they change them (on their User or Player entity) over time, the tournament should remain what it was, as a historical record.

Might be worth updating the notes here w/ that info :shrug:

Reply:
I agree with this information, but there's the small caveat that from when someone registers for a Tournament, they can change their prefix separately but they can't change their gamertag, if I recall correctly, when they go to their tournament registration details. 

This is what I was trying to highlight, actually -- it's the thing that it's great for historical references, but there is also a weird middle ground space between "tournament registration" and "tournament completion" (or even touranment start). 

I also realized I'm not sure how this works with Teams' implicitly defined names, since those seem to have the prefix baked into the names? I don't have the time at the moment to mock a tournament with a team to test this but am curious of the below scenario:

1. User has a prefix on their User profile settings. (Example: `C9`)
2. User registers for a tournament, with their prefix set from their profile (`C9`), for Melee Singles.
3. User registers for Melee Doubles with `C9` prefix, with a teammate that has no prefix. That Team entity has a name of `C9 Mang0 | Polish`.
4. User updates their tournament registration to be `MM`.
5. User updates their global prefix to be `XX`. 

Questions:
1. After 4, does the Team Entrant get updated to be `MM Mang0 | Polish`?
2. How would this be different for a global team, or an entity within an ongoing circuit?

Reply:
I agree with this information, but there's the small caveat that from when someone registers for a Tournament, they can change their prefix separately but they can't change their gamertag, if I recall correctly, when they go to their tournament registration details. 

This is what I was trying to highlight, actually -- it's the thing that it's great for historical references, but there is also a weird middle ground space between "tournament registration" and "tournament completion" (or even touranment start). 

I also realized I'm not sure how this works with Teams' implicitly defined names, since those seem to have the prefix baked into the names? I don't have the time at the moment to mock a tournament with a team to test this but am curious of the below scenario:

1. User has a prefix on their User profile settings. (Example: `C9`)
2. User registers for a tournament, with their prefix set from their profile (`C9`), for Melee Singles.
3. User registers for Melee Doubles with `C9` prefix, with a teammate that has no prefix. That Team entity has a name of `C9 Mang0 | Polish`.
4. User updates their tournament registration to be `MM`.
5. User updates their global prefix to be `XX`. 

Questions:
1. After 4, does the Team Entrant get updated to be `MM Mang0 | Polish`?
2. How would this be different for a global team, or an entity within an ongoing circuit?

Reply:
I'm not sure off-hand what happens, and also don't have the time (just found this tab in my 100+ tab browser..) but in general I think we "try to keep things for historical purposes", so changes only apply to in-progress or future tournaments. Global teams are more of a living entity (just like User) so they should update when you update an event-level team for an in-progress or future event iirc.